### PR TITLE
(proof-new) Proofs for sets purification lemmas

### PR DESCRIPTION
**Pull request by ajreynol**
This adds proofs for sets purification lemmas, which are of the form `(= t (skolem t))` and `(member t (skolem (singleton t)))`.  Each can be trivially justified in the internal calculus by `MACRO_SR_PRED_INTRO`.

### CODE REVIEW

File: src/theory/sets/term_registry.cpp [hunk 1]
```diff
@@ -152,6 +154,21 @@ void TermRegistry::debugPrintSet(Node s, const char* c) const
   }
 }
 
+void TermRegistry::sendSimpleLemmaInternal(Node eq, InferenceId id)
```
Comment (mudathirmahgoub) [inline src/theory/sets/term_registry.cpp:157]: The node argument here could be an equality node, or a singleton node. Please rename the argument to `n`

File: src/theory/sets/skolem_cache.cpp [hunk 2]
```diff
@@ -35,8 +35,19 @@ Node SkolemCache::mkTypedSkolemCached(
...
+    {
+      Assert(a.getType() == tn);
+      sk = sm->mkPurifySkolem(a, c);
+    }
+    else
+    {
+      sk = sm->mkDummySkolem(c, tn, "sets skolem");
+    }
     d_skolemCache[a][b][id] = sk;
+    d_allSkolems.insert(sk);
```
Comment (mudathirmahgoub) [inline src/theory/sets/skolem_cache.cpp:50]: Is this needed? It seems this was working before.  The function `isSkolem` which uses this private field `d_allSkolems` is not used anywhere.

**Review by mudathirmahgoub — APPROVED** [review]
LGTM. It seems we don't need the private field `d_allSkolems`, but its existence only incurs memory cost.

File: src/theory/sets/skolem_cache.cpp (hunk 2 above)
Comment (ajreynol) [reply src/theory/sets/skolem_cache.cpp:50]: isSkolem is used in sets solver state still.

I initially thought this wasnt needed but this does impact how the skolems are handled.

**Pull request merged**